From 48fbbf6e6756584566ae64fbd2c690aeaa7f3efb Mon Sep 17 00:00:00 2001
From: stephan-xyz <stephan@zehrer.xyz>
Date: Tue, 27 Oct 2026 13:58:58 +0100
Subject: [PATCH] Start bcm2835-sdhostd at boot instead of bcm2835-sdhcid

The card slot goes to the sdhost controller; sdhostd serves it as
disk.mmc, so everything that mounts from disk.mmc stays the same.
---
 .../{40_bcm2835-sdhcid.service => 40_bcm2835-sdhostd.service} | 4 ++--
 init.initfs.d/40_drivers.target                               | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)
 rename init.initfs.d/{40_bcm2835-sdhcid.service => 40_bcm2835-sdhostd.service} (71%)

diff --git a/init.initfs.d/40_bcm2835-sdhcid.service b/init.initfs.d/40_bcm2835-sdhostd.service
similarity index 71%
rename from init.initfs.d/40_bcm2835-sdhcid.service
rename to init.initfs.d/40_bcm2835-sdhostd.service
index 8c05462..c494796 100644
--- a/init.initfs.d/40_bcm2835-sdhcid.service
+++ b/init.initfs.d/40_bcm2835-sdhostd.service
@@ -1,9 +1,9 @@
 [unit]
-description = "BCM2835 SD card driver"
+description = "BCM2835 SD card driver (sdhost)"
 requires_weak = ["10_inputd.service", "10_lived.service", "20_graphics.target"]
 condition_architecture = ["aarch64"]
 condition_board = ["raspi3bp"]
 
 [service]
-cmd = "bcm2835-sdhcid"
+cmd = "bcm2835-sdhostd"
 type = "notify"
diff --git a/init.initfs.d/40_drivers.target b/init.initfs.d/40_drivers.target
index d9e2dab..8099bbe 100644
--- a/init.initfs.d/40_drivers.target
+++ b/init.initfs.d/40_drivers.target
@@ -4,7 +4,7 @@ requires_weak = [
     "10_lived.service",
     "20_graphics.target",
     "40_ps2d.service",
-    "40_bcm2835-sdhcid.service",
+    "40_bcm2835-sdhostd.service",
     "40_dwc2d.service",
     "40_hwd.service",
     "40_pcid-spawner-initfs.service",
-- 
2.54.0 (Apple Git-157)

